dispatcher_plugin_nb2workflow/dataserver_dispatcher.py

```diff
@@ -98,6 +98,52 @@ def test_has_input_products(self, instrument, logger=None):
         query_out.set_done('input products check skipped')
         return query_out, []
 
+    def get_progress_run(self,
+                         call_back_url=None,
+                         run_asynch=None,
+                         logger=None,
+                         task=None,
+                         param_dict=None):
+
+        query_out = QueryOutput()
+        res_trace_dict = None
+
+        if task is None:
+            task = self.task
+        if param_dict is None:
+            param_dict = self.param_dict
+        if run_asynch is not None and run_asynch and call_back_url is not None:
+            param_dict['_async_request_callback'] = call_back_url
+            param_dict['_async_request'] = "yes"
+
+        url = os.path.join(self.data_server_url, 'api/v1.0/get', task.strip('/'))
+        res = requests.get(url, params=param_dict)
+        if res.status_code in [200, 201]:
+            res_data = res.json()
+            workflow_status = res_data['workflow_status'] if run_asynch else 'done'
+            if workflow_status == 'started' or workflow_status == 'done':
+                resroot = res_data['data'] if run_asynch and workflow_status == 'done' else res_data
+                jobdir = resroot['jobdir'].split('/')[-1]
+                trace_url = os.path.join(self.data_server_url, 'trace', jobdir, task.strip('/'))
+                res_trace = requests.get(trace_url)
+                if res_trace_dict is None:
+                    res_trace_dict = {}
+                res_trace_dict['res'] = res_trace
+                res_trace_dict['progress_product'] = True
+
+            query_out.set_status(0, job_status=workflow_status)
+        else:
+            try:
+                query_out.set_failed('Error in the backend',
+                                     message='connection status code: ' + str(res.status_code),
+                                     extra_message=res.json()['exceptions'][0])
+            except:
```

Comment:
please capture Exception, report in sentry, in log

Reply:
sentry is not used within this plugin...or at least I haven't seen it

Reply:
in any case, a sentry is sent from the `set_query_exception` method from the dispatcher, with the details of the exception

Reply:
through failed state of query? Then please still catch the exception and provide it to the structure which will be eventually sent to sentry. Right now you are just ignoring some exceptions, `except:` is basically never a good thing.

Reply:
that except was there in case the response from the backend was not json-formatted, I will re-write it to make it cleaner

Reply:
it's better now....

Reply:
> through failed state of query? 

yes, here: https://github.com/oda-hub/dispatcher-app/blob/7257b06ebe157878091ad139eb59361acb65c467/cdci_data_analysis/analysis/products.py#L165-L220